src/caf/base/data_structures.py

```diff
@@ -1536,18 +1572,43 @@ def validate_ipf_targets(
         """
         target_sum = 0.0
         for position, target in enumerate(targets):
-            # Check targets sum to the same, or they can't converge. Potentially could allow
-            # IPF for non-agreeing targets to get as close as possible.
-            if target_sum == 0:
-                target_sum = target.data.sum()
+            subsets = target.data.segmentation.input.subsets
+            if len(subsets) > 0:
+                for comp_target in targets:
+                    if target == comp_target:
+                        continue
+                    # Check that subsets exist in comp_target
+                    if set(subsets.keys()).intersection(
+                        comp_target.data.segmentation.names
+                    ) != set(subsets.keys()):
```

Comment:
I think this could be rewritten as a set comparison which checks if every element in set on the left is in set on the right
```suggestion
                    if set(subsets.keys()) <= set(
                        comp_target.data.segmentation.names
                    ):
```